### PR TITLE
feat: add lib and es6 bundles, use closure compiler, remove gulp

### PR DESCRIPTION
**Pull request by frederickfogerty**
## Description

This PR improves the build step of this library. The minified JS build is reduced from `14K -> 10K = 30% smaller`. The css size is unchanged.

There have been some changes to the source code to support the closure compiler. There might be a chance that things that worked before may not work now due to the optimisations it performs. I have tried to cover all common usage, but there may be some edge cases which I have not tested. All the tests still pass, and I've ensured that the tests test the compiled code.

Gulp has also been removed in favour of npm scripts.

### New Feature

- [x] If this is a big feature with breaking changes, consider [opening an issue][issues] to discuss first. This is completely up to you, but please keep in mind that your PR might not be accepted.
- [x] Run unit tests to ensure all existing tests are still passing
- [x] Add new passing unit tests to cover the code introduced by your PR
- [x] Update the readme
- [x] Add some [steps](#steps-to-test) so we can test your cool new feature!
- [x] The PR title is in the [conventional commit](#conventional-commit-spec) format: e.g. `feat(<area>): added new way to do this cool thing #issue-number`
- [x] Add your info to the [contributors](#packagejson-contributors) array in package.json!

## Steps to Test

1. Clone this PR and run `npm install` to update dependencies
2. Run `npm run build` and open `index.html` in a browser, and test that Luminous still works normally
3. Create a consuming package, npm link, and ensure that Luminous works when importing using `import {Luminous} from 'luminous'`

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -19,14 +19,16 @@
   "homepage": "https://github.com/imgix/luminous",
   "moduleType": [],
   "ignore": [
-    "**/.*",
+		"**/.*",
```
Comment (jayeb) [inline bower.json]: [nit] There's a tabs/spaces mismatch in this file.
